src/numeric/grids.jl

```diff
@@ -5,19 +5,48 @@ function mlinspace(min, max, n)
     return QE.gridmake(nodes...)
 end
 
-type CartesianGrid <: Grid
-    min::Array{Float64,1}
-    max::Array{Float64,1}
-    n::Array{Int64,1}
+type EmptyGrid <: Grid
+    # this grid does not exist ;-)
 end
 
+nodes(grid::EmptyGrid) = nothing
+n_nodes(grid::EmptyGrid) = 0
+node(grid::EmptyGrid, i::Int64) = nothing # fail if i!=1 ?
 
-type SmolyakGrid <: Grid
-    min::Array{Real,1}
-    max::Array{Real,1}
-    mu::Array{Int,1}
+
+type PointGrid <: Grid
+    point::Vector{Float64}
+end
+nodes(grid::PointGrid) = grid.point[:,:]
+n_nodes(grid::PointGrid) = 1
+node(grid::PointGrid, i::Int64) = point # fail if i!=1 ?
+
+
+type UnstructuredGrid <: Grid
+    nodes::Matrix{Float64}
 end
+nodes(grid::UnstructuredGrid) = grid.nodes
+n_nodes(grid::UnstructuredGrid) = size(grid.nodes,1)
+node(grid::UnstructuredGrid, i::Int64) = grid.nodes[i,:] # fail if i!=1 ?
 
-function nodes(grid::CartesianGrid)
-    return mlinspace(grid.min, grid.max, grid.n)
+
+type CartesianGrid <: Grid
+    min::Vector{Float64}
+    max::Vector{Float64}
+    n::Vector{Int64}
+    nodes::Matrix{Float64}
+    function CartesianGrid(min, max, n)
+        nodes = mlinspace(min, max, n)
+        return new(min,max,n,nodes)
+    end
+end
+nodes(grid::Grid) = grid.nodes
+n_nodes(grid::Grid) = size(grid.nodes,1)
+node(grid::Grid, i::Int64) = grid.nodes[i,:]
+
+type SmolyakGrid <: Grid
```

Comment:
Let's consider it as non blocking until we are happy with the overall design. As an interim measure, I guess we can still rely on some methods from Smolyak.jl or include them temporarly.
I would say it is still medium/medium-high priority.